src/VisualStudio/Core/Def/Implementation/LanguageClient/VisualStudioInProcLanguageClient.cs

```diff
@@ -17,19 +17,19 @@
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageClient
 {
     /// <summary>
-    /// Language client responsible for exposing pull diagnostics.  Expected to run in all scenarios where the host
-    /// supports pull diagnostics (including VS and liveshare).
+    /// Language client responsible for C# / VB LSP support locally (outside of liveshare / codespaces).
     /// </summary>
     [ContentType(ContentTypeNames.CSharpContentType)]
     [ContentType(ContentTypeNames.VisualBasicContentType)]
     [Export(typeof(ILanguageClient))]
-    internal class PullDiagnosticsInProcLanguageClient : AbstractInProcLanguageClient
+    [Export(typeof(VisualStudioInProcLanguageClient))]
+    internal class VisualStudioInProcLanguageClient : AbstractInProcLanguageClient
```

Comment:
happy to have suggestions for better names 😆

Reply:
could we just make a separate inprocclient for ctrl-q?

Reply:
I don't really see a reason to make it separate, and I see a few good reasons not to:
1.  Each client creates a new LSP server instance.  This has a small bit of overhead right now (its own execution queue), each one has duplicated initialize/shutdown work.  It gets worse though-
2.  Once @davidwengier change goes in for text document sync, then this overhead becomes fairly large as each server now stores the current document text for any documents opened for that LSP server.  
3.  It makes diagnosing issues a little hard as all the features are spread out across many different language servers.  IMHO we already have too many, so figuring out which server is answering what requests is hard.  I think our goal should be to minimize the number of LSP servers we have, and only create new ones when the scenarios / lifetimes they run in are different.  After this PR we'll have RazorLSPserver (maybe we can delete once pull diag goes on by default), Liveshare (used in the liveshare scenario), and this one (used in local scenario + liveshare).  I would argue against having more.

Reply:
A server-per-function doesn't make sense to me either, as it adds overhead to the platform creating all of these instances, and the user waiting for more messages to fly around the system (eg, the client has to send didChange messages to all servers before it can ask one of them for diagnostics/symbols/etc). The only benefit would be if we had a particularly fragile handler and we didn't want the server to crash all the time, but I'd rather exile the bad handler to its own server (well, obviously I'd rather the situation didn't arise in the first place :P)